src/app/models/mod.rs

```diff
@@ -0,0 +1,181 @@
+use mongodb::bson::{oid::ObjectId, DateTime};
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct User {
+    #[serde(rename = "_id", skip_serializing_if = "Option::is_none")]
+    id: Option<ObjectId>,
+    first_name: String,
+    last_name: String,
+    national_health_identifer: String,
+    email_address: String,
+    hashed_password: String,
+    is_patient: bool,
+    caregivers: Vec<ObjectId>,
+    form_templates: Vec<Form>,
+    form_filled: Vec<Form>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct Form {
+    #[serde(rename = "_id", skip_serializing_if = "Option::is_none")]
+    id: Option<ObjectId>,
+    title: String,
+    created_by: ObjectId,
+    created_at: DateTime,
+    questions: Vec<Question>,
+    events: Vec<Event>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub enum Event {
+    FormSubmitted(FormSubmitted),
+    QuestionEdited(QuestionEdited),
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct QuestionEdited {
```

Comment:
How will questions being added/removed (so that the total number of questions changes) work?

If a given question is edited in such a way that it has a different semantic meaning (i.e. we just replace it entirely), is there any reason to retain the same id rather than just treating it as a deletion + a creation? Is this intended specifically for cases where questions are edited while keeping the same meaning (e.g. typo corrections, or rephrasing the same general question, etc)?